lib/db/mysql/scripts/purge-expired-tokens.sql

```diff
@@ -0,0 +1,43 @@
+DROP PROCEDURE IF EXISTS PURGE_EXPIRED_TOKENS;
+
+DELIMITER $$
+# Create a procedure that will delete a total of `roundCount` * `limitDeletePerRound` items, every `delaySeconds`.
+# Skip tokens that belong clientId
+CREATE PROCEDURE PURGE_EXPIRED_TOKENS(IN roundCount INT, IN limitDeletePerRound INT, IN delaySeconds INT, IN ignoreClientId CHAR(10))
+
+BEGIN
+  DECLARE startIndex INT DEFAULT 1 ;
+  DECLARE endCount INT DEFAULT roundCount ;
+  DECLARE expiredItemCount INT DEFAULT 0;
+
+  PURGE_LOOP: LOOP
+
+	# Exit loop if no expired items found
+    SET expiredItemCount = (SELECT COUNT(*) FROM tokens WHERE expiresAt < NOW() AND UNHEX(clientId) != ignoreClientId ORDER BY expiresAt);
+    IF expiredItemCount = 0 THEN
+		LEAVE PURGE_LOOP;
+	END IF;
+
+	# Perform a round of expired token deletions
+	SET SQL_SAFE_UPDATES = 0;
+	DELETE p1 FROM `fxa_oauth`.`tokens` AS p1
+	INNER JOIN (
+		SELECT token FROM `fxa_oauth`.`tokens` WHERE expiresAt < NOW() AND UNHEX(clientId) != ignoreClientId ORDER BY expiresAt ASC LIMIT limitDeletePerRound
```

Comment:
I like that idea. As a side note, do you think it makes more sense for the procedure to delete tokens from a specific clientId instead of all clients but ignoring a single clientId? Seems like we could have a little more flexibility this way.